libc/main.c

```diff
@@ -39,13 +39,80 @@ void runner(void)
 	RUN_TEST_GROUP(unistd_file);
 	RUN_TEST_GROUP(wchar_wcscmp);
 	RUN_TEST_GROUP(test_pthread_cond);
+}
+
+
+/* crete directory unless it exists */
+static int libc_createDirIfMissing(const char *path)
+{
+	struct stat buffer;
 
-	unsetenv("POSIXLY_CORRECT");
+	if (stat(path, &buffer) != 0) {
+		if (errno == ENOENT) {
+			if (mkdir(path, S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH) != 0) {
+				fprintf(stderr, "Creating %s directory by mkdir failed: %s\n", path, strerror(errno));
+				return -1;
+			}
+		}
+		else {
+			fprintf(stderr, "stat() on %s directory failed: %s\n", path, strerror(errno));
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+
+/* crete file with optional data (may be NULL) unless it exists */
+static int libc_createFileIfMissing(const char *path, const char *charData)
+{
+	struct stat buffer;
+
+	if (stat(path, &buffer) != 0) {
+		if (errno == ENOENT) {
+			if (_create_file(path, charData) != 0) {
+				fprintf(stderr, "Creating %s file failed: %s\n", path, strerror(errno));
+				return -1;
+			}
+		}
+		else {
+			fprintf(stderr, "stat() on %s file failed: %s\n", path, strerror(errno));
+			return -1;
+		}
+	}
+
+	return 0;
 }
 
 
 int main(int argc, char *argv[])
 {
+	const char *var = "POSIXLY_CORRECT";
+
+	if (setenv(var, "y", 1) != 0) {
+		fprintf(stderr, "Setting %s environment variable failed: %s\n", var, strerror(errno));
+		return 1;
+	}
+
+	/* the following files may not be present on dummyfd targets,
+	create them to make libc tests common */
+	if (libc_createDirIfMissing("/tmp") < 0) {
+		return 1;
```

Comment:
Sorry, missed that on the previous review. If it is important to `unsetenv` before exit, we should do this on these error exit paths too

Reply:
Yeah, but I think not so important to check its return code every time, so I would prefer sth like this:
```C
int main(int argc, char *argv[])
{
	const char *var = "POSIXLY_CORRECT";

	if (setenv(var, "y", 1) != 0) {
		fprintf(stderr, "Setting %s environment variable failed: %s\n", var, strerror(errno));
		return 1;
	}

	/* the following files may not be present on dummyfd targets,
	create them to make libc tests common */
	if (libc_createDirIfMissing("/tmp") < 0) {
		unsetenv(var);
		return 1;
	}
	if (libc_createDirIfMissing("/etc") < 0) {
		unsetenv(var);
		return 1;
	}
	if (libc_createFileIfMissing("/etc/passwd", "root:0B1ANiYi45IhxkfmUW155/GBd4IRE=:0:0:root:/:/bin/sh")) {
		unsetenv(var);
		return 1;
	}

	UnityMain(argc, (const char **)argv, runner);

	if (unsetenv(var) != 0) {
		fprintf(stderr, "Unsetting %s environment variable failed: %s\n", var, strerror(errno));
		return 1;
	}

	return 0;
}
```